include/OGL/MatrixWrapper/HostMatrix.H

```diff
@@ -0,0 +1,231 @@
+
+    const lduAddressing &addr_;
+
+    const scalar *diag_;
+
+    const scalar *upper_;
+
+    const scalar *lower_;
+
+    // multiply the complete system by this factor, ie sAx=sb
+    // NOTE this needed to avoid negative diagonal matrix entries, but
+    // this could be also achieved by just fliping the sign
+    const scalar scaling_;
+
+    // number of local matrix rows
+    const label nrows_;
+
+    // number of local upper elements
+    // ie coefficients which column_idx < nrows_
+    const label upper_nnz_;
+
+    const bool symmetric_;
+
+    // number of local elements on interfaces
+    // ie number of interfaces which column_idx is < nrows_
+    // and have to be sorted into local matrix
+    const label local_interface_nnz_;
+
+    // total number of local upper and lower elements
+    // ie 2*upper_nnz_ since the sparsity pattern is symmetric
+    const label non_diag_nnz_;
+
+    // nnz of local matrix wo local interfaces
+    const label local_matrix_nnz_;
+
+    // nnzs of local matrix including local interfaces
+    const label local_matrix_w_interfaces_nnz_;
+
+    // non-local indices
+    const label non_local_matrix_nnz_;
+
+    const lduInterfaceFieldPtrsList &interfaces_;
+
+    const FieldField<Field, scalar> &interfaceBouCoeffs_;
+
+    /* Iterates all interfaces and collects the coefficients into a vector
+    **
+    ** @param local whether local or non local coefficients should be collected
+    */
+    [[nodiscard]] std::vector<scalar> collect_interface_coeffs(
+        const lduInterfaceFieldPtrsList &interfaces_,
+        const FieldField<Field, scalar> &interfaceBouCoeffs,
+        const bool local) const;
+
+    /* Iterates all interfaces and counts the number of elements
+    **
+    ** @param interfaces The list of interfaces for the search
+    ** @param proc_interfaces Count only elements on (true)
+    *processorLduInterfaces or exclude processorLduInterfaces (false)
+    * @return number of elements on interfaces
+    */
+    [[nodiscard]] label count_interface_nnz(
+        const lduInterfaceFieldPtrsList &interfaces,
+        bool proc_interfaces) const;
+
+    /** Iterates all local interfaces and returns the relative order and
+    **corresponding row and column indices
+    **
+    ** @return vector of tuples containing the interface number, the local row,
+    *the local column
+    **/
+    std::vector<std::tuple<label, label, label>>
+    collect_local_interface_indices(
+        const lduInterfaceFieldPtrsList &interfaces_) const;
+
+
+    /** Iterates all interfaces and collect the corresponding cell id (row)
+    ** and a unique counter
+    **
+    ** @return vector of size nnz_non_local, with a running index, row
+    ** index, and a unique compressed column index
+    ** sections for each interface
+    ** ret = [(1,2,1),(2,20, 2), (3, 20, 2) ...]
+    **         i0   i1,   i...
+    */
+    [[nodiscard]] std::vector<std::tuple<label, label, label, label>>
```

Comment:
There is a inconsistency between the comment and the function declaration regarding the return value.
In the comment, a vector of 3-element tuples is specified for the return value. However, the function declaration has a vector of 4-element tuples as the return type.

Reply:
good catch.